cpp/include/cuspatial/experimental/detail/indexing/construction/phase_1.cuh

```diff
@@ -58,43 +55,41 @@ namespace detail {
  * @brief Compute Morton codes (z-order) for each point, and the sorted indices
  * mapping original point index to sorted z-order.
  */
-template <typename T>
-inline std::pair<rmm::device_uvector<uint32_t>, std::unique_ptr<cudf::column>>
-compute_point_keys_and_sorted_indices(cudf::column_view const& x,
-                                      cudf::column_view const& y,
-                                      T x_min,
-                                      T x_max,
-                                      T y_min,
-                                      T y_max,
+template <class PointIt, class T>
+inline std::pair<rmm::device_uvector<uint32_t>, rmm::device_uvector<uint32_t>>
+compute_point_keys_and_sorted_indices(PointIt points_first,
+                                      PointIt points_last,
+                                      vec_2d<T> min,
+                                      vec_2d<T> max,
                                       T scale,
                                       int8_t max_depth,
                                       rmm::cuda_stream_view stream,
                                       rmm::mr::device_memory_resource* mr)
 {
-  rmm::device_uvector<uint32_t> keys(x.size(), stream);
-  thrust::transform(rmm::exec_policy(stream),
-                    thrust::make_zip_iterator(x.begin<T>(), y.begin<T>()),
-                    thrust::make_zip_iterator(x.begin<T>(), y.begin<T>()) + x.size(),
-                    keys.begin(),
-                    [=] __device__(auto const& point) {
-                      T x, y;
-                      thrust::tie(x, y) = point;
-                      if (x < x_min || x > x_max || y < y_min || y > y_max) {
-                        // If the point is outside the bbox, return a max_level key
-                        return static_cast<uint32_t>((1 << (2 * max_depth)) - 1);
-                      }
-                      return cuspatial::utility::z_order((x - x_min) / scale, (y - y_min) / scale);
-                    });
-
-  auto indices = make_fixed_width_column<uint32_t>(keys.size(), stream, mr);
-
-  thrust::sequence(rmm::exec_policy(stream),
-                   indices->mutable_view().begin<uint32_t>(),
-                   indices->mutable_view().end<uint32_t>());
+  auto num_points = thrust::distance(points_first, points_last);
+  rmm::device_uvector<uint32_t> keys(num_points, stream);
+  thrust::transform(
+    rmm::exec_policy(stream),
+    points_first,
+    points_last,
+    keys.begin(),
+    [=] __device__(vec_2d<T> const& point) {
+      // auto const& x = point.x;
+      // auto const& y = point.y;
+      if (point.y < min.x || point.y > max.x || point.y < min.y || point.y > max.y) {
```

Comment:
Since this is likely to be a common operation (already exists here: https://github.com/rapidsai/cuspatial/blob/cd0f2b3ef9272839efcdbef489fbd93bd445d402/cpp/include/cuspatial/experimental/detail/points_in_range.cuh#L46), perhaps it should be factored out to minimize mistakes like this.